.github/workflows/scripts/monitor-typescript.ts

```diff
@@ -39,44 +39,52 @@ const ALLOWED_KEYS = new Set([
 
 const tsLogFile = fs.readFileSync('/tmp/typescript-monitor.log', 'utf8');
 
-function toKeyValue(input) {
+function parseContentsToKeyValue(
+  input: string
+): Record<string, {unit: string | undefined; value: number}> | null {
   try {
     return input
       .split('\n')
-      .filter(n => !!n)
-      .reduce((acc, line) => {
-        const [k, v] = line.split(':');
+      .filter(n => !!n.trim())
+      .reduce(
+        (
+          acc: Record<string, {unit: string | undefined; value: number}>,
+          line: string
+        ) => {
+          const [k, v] = line.split(':');
+
+          if (typeof k !== 'string' || typeof v !== 'string') {
+            return acc;
+          }
 
-        if (typeof k !== 'string' || typeof v !== 'string') {
-          return acc;
-        }
+          const key = k.trim();
+          const value = v.trim();
 
-        const key = k.trim();
-        const value = v.trim();
+          if (ALLOWED_KEYS.has(key)) {
+            let number = parseFloat(value);
+            let unit = undefined;
 
-        if (ALLOWED_KEYS.has(key)) {
-          let number = parseFloat(value);
-          let unit = undefined;
+            if (!isNaN(number)) {
+              acc[key] = {value: number, unit};
+            }
 
-          if (!isNaN(number)) {
-            acc[key] = {value: number, unit};
+            if (/([a-zA-Z]$)/.test(value)) {
+              number = parseFloat(v.substring(0, v.length - 2));
+              unit = v.match(/([a-zA-Z]$)/)?.[0];
+              acc[key] = {value: number, unit};
+            }
           }
 
-          if (/([a-zA-Z]$)/.test(value)) {
-            number = parseFloat(v.substring(0, v.length - 2));
-            unit = v.match(/([a-zA-Z]$)/)?.[0];
-            acc[key] = {value: number, unit};
-          }
-        }
-
-        return acc;
-      }, {});
+          return acc;
```

Comment:
```
if (ALLOWED_KEYS.has(key)) {
            const parsedValue = parseFloat(value);

            if (!isNaN(parsedValue)) {
              acc[key] = {
                value: parsedValue,
                unit: undefined,
              };
            }

            if (/([a-zA-Z]$)/.test(value)) {
              acc[key] = {
                value: parseFloat(v.substring(0, v.length - 2)),
                unit: v.match(/([a-zA-Z]$)/)?.[0],
              };
            }
          }
```

🤷‍♀️

Reply:
We can also avoid just reparsing the value in 2nd if block. I'll push a fix